cmd/influx/user_test.go

```diff
@@ -0,0 +1,383 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"testing"
+
+	"github.com/influxdata/influxdb"
+	platform "github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/mock"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	input "github.com/tcnksm/go-input"
+)
+
+func TestCmdUser(t *testing.T) {
+	setViperOptions()
+
+	orgID := influxdb.ID(9000)
+
+	type userResult struct {
+		user     influxdb.User
+		password string
+	}
+
+	fakeSVCFn := func(
+		svc influxdb.UserService,
+		passSVC influxdb.PasswordsService,
+		getPassFn func(*input.UI, bool) string,
+	) userSVCsFn {
+		return func() (
+			influxdb.UserService,
+			influxdb.OrganizationService,
+			influxdb.PasswordsService,
+			influxdb.UserResourceMappingService,
+			func(*input.UI, bool) string,
+			error) {
+			return svc, &mock.OrganizationService{
+					FindOrganizationF: func(ctx context.Context, filter influxdb.OrganizationFilter) (*influxdb.Organization, error) {
+						return &influxdb.Organization{ID: orgID, Name: "influxdata"}, nil
+					},
+				}, passSVC, &mock.UserResourceMappingService{
+					CreateMappingFn: func(context.Context, *platform.UserResourceMapping) error {
+						return nil
+					},
+				}, getPassFn, nil
+		}
+	}
+
+	t.Run("create", func(t *testing.T) {
+		tests := []struct {
+			name     string
+			expected userResult
+			flags    []string
+		}{
+			{
+				name:  "basic just name",
+				flags: []string{"--name=new name", "--org=org name"},
+				expected: userResult{
+					user: influxdb.User{
+						Name: "new name",
+					},
+				},
+			},
+			{
+				name: "with password",
+				flags: []string{
+					"--name=new name",
+					"--password=pass1",
+					"--org=org name",
```

Comment:
can you add some tests for the env var as part of the `--org` and `--org-id` API contract? Here are some examples from the org service that allow for env vars: https://github.com/influxdata/influxdb/blob/master/cmd/influx/organization_test.go#L247